ChildProject/pipelines/metrics.py

```diff
@@ -40,6 +41,8 @@ class Metrics(ABC):
     :type child_cols: str, optional
     :param period: time units to aggregate (optional); equivalent to ``pandas.Grouper`` freq argument.
     :type period: str, optional
+    :param segments: DataFrame or path to csv file of the segments to extract from, containing 'recording_filename', 'segment_onset' and 'segment_offset' columns. To use this option, the option must be set to 'segments'. Also, this option cannot be combined with options [recordings,period,from_time,to_time].
+    :type segments: Union[str, pd.DataFrame], optional
```

Comment:
I think I got lost in the terminology again. I thought that _segments_ referred to intervals with individual annotated utterances while _annotations_ referred to the time intervals that might contain _segments_. But here _segments_ refers to what I thought were _annotations_. Could you please explain the difference to me again?

Reply:
sorry about that, in our terminology, segment = stretch of audio

So you can have an utterance that some human or machine has split up into three segments -- the word segment only means that they have a start & end time, it doesn't tell you that eg they are internally consistent (in the sense that I think you were thinking about, perhaps segment = phone, or segment = a single cry)

AND you can have a recording that has been split into parts, for instance, sections in which humans have carried out annotation